File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    
    nums = []
    for number in numbers: nums.append(str(number))
    
    #number는 1000이하의 수 -> 3번 이어 붙인 값의 크기 기준 내림차순 정렬
    nums.sort(key=lambda x: list(x*3)[:4], reverse=True)
    
    result = ""
    for num in nums: result += "".join(map(str, num))
    return str(int(result))

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<int> numbers) {
    
    sort(numbers.begin(), numbers.end(), [](const int &a, const int &b)
         {
             string sa = to_string(a) + to_string(b); 
             string sb = to_string(b) + to_string(a);
             return sa > sb;
         });
    
    string result = "";
    for (int i = 0; i < numbers.size(); ++i)
    {
        result += to_string(numbers[i]);
    }
    
    if (result[0] == '0')
    {
        return "0";
    }
    return result;
}